protobuf/connectmsg.proto

```diff
@@ -0,0 +1,10 @@
+//package com.bloomberg.comdb2.net;
+syntax = "proto2";
+
+message net_connectmsg {
+  optional string to_hostname = 1;
+  optional int32 to_portnum = 2;
+  optional string my_hostname = 3;
+  optional int32 my_portnum = 4;
+  optional string dbname = 5;
```

Comment:
Why are these optional?

Reply:
From https://protobuf.dev/getting-started/cpptutorial/

"Required Is Forever You should be very careful about marking fields as required. If at some point you wish to stop writing or sending a required field, it will be problematic to change the field to an optional field – old readers will consider messages without this field to be incomplete and may reject or drop them unintentionally. You should consider writing application-specific custom validation routines for your buffers instead. Within Google, required fields are strongly disfavored; most messages defined in proto2 syntax use optional and repeated only. (Proto3 does not support required fields at all.)"

Reply:
I still error in C code if those fields are not present though. Do you think they should be required?

Reply:
If we were to update to proto 3 in the future we would have to use optional anyways

Reply:
Don’t Add a Required Field
Never add a required field, instead add // required to document the API contract. Required fields are considered harmful by so many they were removed from proto3 completely. Make all fields optional or repeated. You never know how long a message type is going to last and whether someone will be forced to fill in your required field with an empty string or zero in four years when it’s no longer logically required but the proto still says it is.

Reply:
https://protobuf.dev/programming-guides/dos-donts/

Reply:
If you still think I should though I'll change it to required

Reply:
I did what the docs suggested and added a // required comment on each of the fields